### PR TITLE
WA groups: Add support to receive text message from groups

### PR DESCRIPTION
**Pull request by madclaws**
## Summary
 Target issue is #3341 

Added the changes needed in handling contacts and messages in the receive flow when maytapi as a provider. 

## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [x] Ran `mix setup` in the repository root and test.
- [x] If you've fixed a bug or added code that is tested and has test cases. 

## Notes
 Please add here if any other information is required for the reviewer.

### CODE REVIEW

**Review by AmishaBisht — APPROVED** [review]

File: lib/glific/communications/message.ex [hunk 1]
```diff
@@ -245,7 +248,7 @@ defmodule Glific.Communications.Message do
   defp receive_text(message_params) do
     message_params
     |> Messages.create_message()
-    |> publish_data(:received_message)
+    |> publish_data(get_received_msg_publish_event(message_params))
```
Comment (AkhileshNegi) [inline lib/glific/communications/message.ex]: get value before passing into the function. looks cleaner

message_event = get_received_msg_publish_event(message_params))
message_params
    |> Messages.create_message()
    |> publish_data(message_event)

File: test/glific_web/providers/gupshup/controllers/message_controller_test.exs [hunk 2]
```diff
@@ -144,6 +144,114 @@ defmodule GlificWeb.Providers.Gupshup.Controllers.MessageControllerTest do
                get_in(message_params, ["payload", "sender", "phone"])
     end
 
+    test "Updating the contacts due to sender contact already existing", %{
+      conn: conn,
+      message_params: message_params
+    } do
+      conn_bak = conn
```
Comment (AkhileshNegi) [inline test/glific_web/providers/gupshup/controllers/message_controller_test.exs]: what's `conn_bak` ? can you use simpler names

File: test/glific_web/providers/gupshup/controllers/message_controller_test.exs [hunk 3]
```diff
@@ -144,6 +144,114 @@ defmodule GlificWeb.Providers.Gupshup.Controllers.MessageControllerTest do
...
+      # handling a message from gupshup, so that the phone number will be already existing
+      # in contacts table.
+      conn = post(conn, "/gupshup", message_params)
```
Comment (AkhileshNegi) [inline test/glific_web/providers/gupshup/controllers/message_controller_test.exs]: are we sending in `/gupshup` or `/maytapi`

File: test/glific_web/providers/maytapi/controllers/message_controller_test.exs [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+      }
+    }
+  }
+
+  @text_message_webhook %{
+    "product_id" => "5351f38b-c0ae-49c4-9e43-427cb901b0f7",
+    "phone_id" => 42_908,
+    "message" => %{
+      "type" => "text",
+      "text" => "It's like a mini-sprint- Almost half of the team is there",
```
Comment (AkhileshNegi) [inline test/glific_web/providers/maytapi/controllers/message_controller_test.exs]: use generic message and phone number(maybe one that are already in the seeder)

File: test/glific_web/providers/maytapi/controllers/message_controller_test.exs [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+      "_serialized" => "false_120363027326493365@g.us_3EB037B863B86D2AF69DD8_919642961343@c.us",
+      "fromMe" => false
+    },
+    "user" => %{
+      "id" => "919917443994@c.us",
+      "name" => "user_a",
+      "phone" => "919917443994"
+    },
+    "conversation" => "120363213149844251@g.us",
+    "conversation_name" => "Tech4Dev Team",
```
Comment (AkhileshNegi) [inline test/glific_web/providers/maytapi/controllers/message_controller_test.exs]: use generic names and omit `Tech4Dev`

**Review by AkhileshNegi — CHANGES_REQUESTED** [review]

**Review by AkhileshNegi — APPROVED** [review]

**Pull request merged**